modules/backend/widgets/Filter.php

```diff
@@ -380,19 +380,16 @@ protected function filterActiveOptions(array $activeKeys, array &$availableOptio
 
     /**
      * Looks at the model for defined scope items.
+     * @param \Backend\Classes\FilterScope $scope
+     * @param string $searchQuery
      * @return Collection
      */
     protected function getOptionsFromModel($scope, $searchQuery = null)
     {
         $model = $this->scopeModels[$scope->scopeName];
 
         $query = $model->newQuery();
-
-        /*
-         * The 'group' scope has trouble supporting more than 500 records at a time
-         * @todo Introduce a more advanced version with robust list support.
-         */
-        $query->limit(500);
+        $query->limit(100);
```

Comment:
@danielbidala Would you be able to explain how dropping the limit to 100 is going to help with the original issue? Wouldn't this just cut off items after 100?

Reply:
@bennothommo When widget loads it gets the first 100 record from the model and populates `scopeAvailable` js array with items on client side and renders `<ul>`.
When you select or deselect an item the `selectItem` method in `filter.js` checks whether the selected item is in the `scopeValues` (active) or `scopeAvailable` array and moves it to the opposite.

**Till now:** When you searched in the group filter another query ran and the <ul> was refreshed with search results but the `scopeAvailable` array was not refreshed.

**With this PR:** If the selected item is not in any of the arrays (`scopeValues` or `scopeAvailable`), `selectItem` method will push it to the right place and you can apply the filter. When page reloads `getOptionsFromModel` in `Filter.php` checks if there are active filters in this scope and if yes it runs additional query to get those records from model and merges the result with the base `$query->limit(100)`. So if you have an active filter which is not in the first 100, it will be in `scopeAvailable` array anyway.

Thanks to this maybe we can limit the query to 50 instead of 100, because the user won't scroll 100 option, he will serach instead. What's your opinion?

Reply:
@LukeTowers I added a new commit according to your suggestion.